packages/edit-site/src/components/block-editor/inserter-media-categories.js

```diff
@@ -0,0 +1,160 @@
+	}
+	// If version is known, append version to the name.
+	// The license has to have a version to be valid. Only
+	// PDM (public domain mark) doesn't have a version.
+	if ( licenseVersion ) {
+		licenseName += ` ${ licenseVersion }`;
+	}
+	// For licenses other than public-domain marks, prepend 'CC' to the name.
+	const isPublicDomainMark = [ 'pdm', 'cc0' ].includes( license );
+	if ( ! isPublicDomainMark ) {
+		licenseName = `CC ${ licenseName }`;
+	}
+	if ( !! licenseUrl ) {
+		license = getExternalLink(
+			`${ licenseUrl }?ref=openverse`,
+			licenseName
+		);
+	}
+	const markedLicence = isPublicDomainMark
+		? sprintf(
+				/* translators: %s: Name of the media's license (ex. 'CC0 1.0'). */
+				__( 'is marked with %s' ),
+				license
+		  )
+		: sprintf(
+				/* translators: %s: Name of the media's license (ex. 'CC BY-ND 2.0'). */
+				__( 'is licensed under %s' ),
+				license
+		  );
+	return markedLicence;
+};
+
+const getOpenverseCaption = ( item ) => {
+	const {
+		title,
+		foreign_landing_url: foreignLandingUrl,
+		creator,
+		creator_url: creatorUrl,
+		license,
+		license_version: licenseVersion,
+		license_url: licenseUrl,
+	} = item;
+	/* translators: Openverse default media item title in the block inserter's media list. (ex. 'This work by {creator} is marked with CC0 1.0'). */
+	const defaultTitle = __( 'This work' );
+	let _title = title ? decodeEntities( title ) : defaultTitle;
+	_title = getExternalLink( foreignLandingUrl, _title );
+	const fullLicense = getOpenvereLicense(
+		license,
+		licenseVersion,
+		licenseUrl
+	);
+	let _creator = decodeEntities( creator );
+	if ( creatorUrl ) {
+		_creator = getExternalLink( creatorUrl, creator );
+	}
+	_creator = sprintf(
+		/* translators: %s: Name of the media's creator. */
+		__( ' by %s' ),
+		_creator
+	);
+	const caption = `"${ _title }" ${ _creator } ${ fullLicense }.`;
+	return caption.replace( /\s{2}/g, ' ' );
+};
+
+const coreMediaFetch = async ( query = {} ) => {
+	const mediaItems = await resolveSelect( coreStore ).getMediaItems( {
+		...query,
+		orderBy: !! query?.search ? 'relevance' : 'date',
+	} );
+	return mediaItems.map( ( mediaItem ) => ( {
+		...mediaItem,
+		alt: mediaItem.alt_text,
+		url: mediaItem.source_url,
+		caption: mediaItem.caption?.raw,
+	} ) );
+};
+
+const inserterMediaCategories = [
+	{
+		label: __( 'Images' ),
+		name: 'images',
+		mediaType: 'image',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'image' } );
+		},
+	},
+	{
+		label: __( 'Videos' ),
+		name: 'videos',
+		mediaType: 'video',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'video' } );
+		},
+	},
+	{
+		label: __( 'Audio' ),
+		name: 'audio',
+		mediaType: 'audio',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'audio' } );
+		},
+	},
+	{
+		label: 'Openverse',
+		name: 'openverse',
+		mediaType: 'image',
```

Comment:
Openverse currently supports images and audio and will one day support everything from video, 3d models and text. Some of those, I imagine, might not make sense to include in the WP media inserter, but audio and video might. Is the current implementation extensible enough to allow for that? My brief look at this would make me think there'd end up being an "Openverse Images" and "Openverse Audio" categories, etc... I guess the media inserter doesn't currently mix media in the same view maybe?

Not a blocker or anything, just wondering if it was a consideration or if it would be fine anyway.

Reply:
Yeah I know, I think we'll explore this when the time comes as there is not even a design for this and as you said currently we don't mix the media types in the inserter. In theory I think we'd make the `mediaType` an array and make changes accordingly.